src/main/java/org/broadinstitute/hellbender/engine/spark/SAMRecordToGATKReadAdapterSerializer.java

```diff
@@ -86,12 +82,12 @@ public LazyBAMRecord(
 
         @Override
         public void setReferenceIndex(final int referenceIndex) {
-            mReferenceIndex = referenceIndex;
+            mReferenceIndex = null; // null out
```

Comment:
I find this is a bit icky. I think we should log a warning here because this is would be surprising behavior for clients.